### ISSUE TITLE
HTS model

### ISSUE THREAD

**Issue by elnursamedzade** [Olga-Yakovleva/RHVoice#191]
Hello again.
Thank you for coming back to me.
Got it, we can't use MaryTTS voices with RHVoice.
So if I train my own model with HTS, how do I integrate it into RHVoice? Or is it possible to train with RHVoice?
I want to contribute to the Azerbaijani language. I hope you don't hesitate to help me.
Thank you.

**Comment by Olga-Yakovleva** [comment]
You'd need to implement the language support first, such as 
pronunciation rules etc. As I've already mentioned in other similar 
issues here, currently it's not easy. The only way to figure it out is 
to study my source code and learn to use Foma, which I'm using a lot to 
implement g2p and other rules. I'm working on a rewrite of the system to 
make it possible for other people, not just myself, to work on new 
languages, but that project is going to be a work-in-progress for some 
time yet. But I can't see any other way to solve this problem.

**Comment by elnursamedzade** [comment]
Yes, I understand you, Olga.
I edited the foma files in my own language and rebuild as fst.
What should I do next? Can you light me up a little more?
Just give any hint.
Besides, you are doing a very good job. Dude you are great. I love you!

**Comment by Olga-Yakovleva** [comment]
There are other files you may study in other languages: they define 
phonemes, hts label format, the alphabet is defined in language-specific 
C++-code.

**Comment by elnursamedzade** [comment]
Yes, I saw those files in the data folder. As far as I understand, I need to convert hts lab files to labelling.xml file with c ++ code.
How can I do that? Is there such a C ++ module in the RHVoice directory?

**Comment by Olga-Yakovleva** [comment]
See such examples as english.hpp/english.cpp, russian.hpp/russian.cpp 
etc. As for labels, the file just defines the format and which feature 
functions to use. You may implement custom feature functions in 
<yourlanguage>.cpp. It all depends on what information it is possible to 
deduce from the text and what can be used to predict which specific 
allophone of a phoneme will be used. It all depends on your specific 
language, so you need to decide what may be important to help the system 
make better models. Such guesses may be wrong, but HTS should be able to 
figure out which of the provided features are actually relevant.

**Comment by elnursamedzade** [comment]
Ok darling, I'll review the files again.
Thank you very much.

**Comment by abitrolly** [comment]
I found a really good HTS tutorial at http://www.cs.columbia.edu/~ecooper/tts/ As `RHVoice` is based on HTS, I wonder how far the process differs from the what is described there?

**Comment by abitrolly** [comment]
There is also a tool called [aeneas](https://github.com/readbeyond/aeneas) that can help with automatic labelling in different languages.
